go/arrow/csv/transformer.go

```diff
@@ -225,6 +227,52 @@ func (w *Writer) transformColToStringArr(typ arrow.DataType, col arrow.Array) []
 				res[i] = w.nullValue
 			}
 		}
+	case arrow.ExtensionType:
+		arr := col.(array.ExtensionArray)
+		b, err := arr.MarshalJSON()
+		if err != nil {
+			panic(fmt.Errorf("arrow/csv: could not marshal extension array: %w", err))
+		}
+		var stringArr []interface{}
+		if err := json.Unmarshal(b, &stringArr); err != nil {
+			panic(fmt.Errorf("arrow/csv: could not unmarshal extnesion to string array: %s", err))
+		}
```

Comment:
why marshal to json and then back? 

I'd much prefer a new method added to an ExtensionType or the `ExtensionArray` interface that implements `Value(int) fmt.Stringer` to allow customized retrieval as whatever type is desired and requiring it to have a `String() string` method so it can just be printed out here.